File: LeetCode/June Challenge/04-Reverse String/solution.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        i, j = 0, len(s) - 1
        while i < j:
            if s[i] == s[j] :
                i += 1
                j -= 1
            else:
                s[i], s[j] = s[j], s[i]
                i += 1
                j -= 1
        

File: LeetCode/June Challenge/04-Reverse String/solution.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        if(!s.size())
            return;
        for(int i=0,j=s.size()-1;i<=s.size()/2 && j>=s.size()/2;i++,j--)
            swap(s[i],s[j]);
    }
};
